lighthouse-plugin-publisher-ads/utils/resource-classification.js

```diff
@@ -94,6 +84,91 @@ function isGptAdRequest(request) {
   );
 }
 
+/**
+ * @param {Artifacts['IFrameElement']} iframe
+ * @return {boolean}
+ */
+function isGptIframe(iframe) {
+  return /(^google_ads_iframe_)/.test(iframe.id);
+}
+
+/**
+ * Checks if the url is loading an AdSense script.
+ * @param {URL|string} url
+ * @return {boolean}
+ */
+function isAdSenseTag(url) {
+  const {host, pathname} = toURL(url);
+  const matchesHost = [
+    'pagead2.googlesyndication.com'].includes(host);
+  const matchesPath =
+      ['/pagead/js/adsbygoogle.js'].includes(pathname);
+  return matchesHost && matchesPath;
+}
+
+/**
+ * Checks if the url is loading an AdSense script. Kind of pointless since
+ * AdSense doesn't have a separate loader, but provided anyway for symmetry with
```

Comment:
AdSense has two loaders! adsbygoogle (modern, preferred) and show_ads.js (deprecated, sync only). The impl is show_ads_impl.js

Reply:
Whoa...I had no idea. Thanks!

Reply:
Done